src/ui/localization/translator.ts

```diff
@@ -0,0 +1,114 @@
+import en from "./locale/en.json";
+import es from "./locale/es.json";
+import it from "./locale/it.json";
+import { PeriodUnit } from "../../helpers/duration-helper";
+
+export class Translator {
+  private readonly locales: Record<string, LocaleTranslations> = {};
+
+  private static instance: Translator;
+
+  public static getInstance(): Translator {
+    if (!Translator.instance) {
+      Translator.instance = new Translator();
+    }
+    return Translator.instance;
+  }
+
+  private constructor() {
+    this.locales = {
+      en: new LocaleTranslations(en),
+      es: new LocaleTranslations(es),
+      it: new LocaleTranslations(it),
+    };
+  }
+
+  public override(customTranslations: Record<string, Record<string, string>>) {
+    Object.entries(customTranslations).forEach(([locale, translations]) => {
+      this.locales[locale] = new LocaleTranslations({
+        ...(this.locales[locale].labels || {}),
+        ...translations,
+      });
+    });
+  }
+
+  private getLocaleInstance(locale: string): LocaleTranslations | undefined {
+    const potentialLocaleCode = locale.split("_")[0].split("-")[0];
+    return (
+      Translator.getInstance().locales[locale] ||
+      Translator.getInstance().locales[potentialLocaleCode]
+    );
+  }
+
+  public static translate(
```

Comment:
This is translating a label, we could call it `translateLabel`.
However I'd like to avoid having `translatePeriodLabel`. I think translate it's fine.